src/main/resources/assets/computercraft/lua/rom/programs/shell.lua

```diff
@@ -114,7 +115,22 @@ else
 end
 
 local function run( _sCommand, ... )
-    local sPath = shell.resolveProgram( _sCommand )
+    local checkvar = { ... }
+    table.insert( checkvar, 1, _sCommand )
+    for k, v in ipairs( checkvar ) do
+        if v:find( "$", 1, true ) == 1 then
+            v = v:sub( 2, -1)
+            if type( shellenv[v] ) == "string" or type( shellenv[v] ) == "number" or type( shellenv[v] ) == "boolean" then
+                v = tostring( shellenv[v] )
+            elseif type( shellenv[v] ) == "function" then
+                v = tostring( shellenv[v]() )
```

Comment:
I feel this would be more elegantly handled via [command substitution](http://tldp.org/LDP/abs/html/commandsub.html). So `foo $(time)` would execute the `time` command and pass its output as an argument to `foo`. If we go this route, then you can also implement the various pipes (`>`, `<`, `|`).